src/renderer/containers/Settings/NetworkDiagnostic/NetworkDiagnosticsButton.tsx

```diff
@@ -0,0 +1,186 @@
+    case NatType.FAILED:
+      return `Please try again later.
+        If the failure persists, this network may block UDP.
+        In that case it will not be possible to use Slippi as well as many other online games and apps.`;
+    default:
+      return "";
+  }
+};
+
+const getPortMappingTitle = (portMapping: PortMapping) => {
+  if (portMapping.upnp === Presence.FAILED || portMapping.natpmp === Presence.FAILED) {
+    return "Failed to determine port mapping availability";
+  }
+  return "Port mapping";
+};
+
+const getPortMappingDescription = (portMapping: PortMapping) => {
+  if (portMapping.upnp === Presence.ABSENT && portMapping.natpmp === Presence.ABSENT) {
+    return "Not available.";
+  }
+  if (portMapping.upnp === Presence.PRESENT && portMapping.natpmp === Presence.ABSENT) {
+    return "Available - UPnP.";
+  }
+  if (portMapping.upnp === Presence.ABSENT && portMapping.natpmp === Presence.PRESENT) {
+    return "Available - NAT-PMP.";
+  }
+  if (portMapping.upnp === Presence.PRESENT && portMapping.natpmp === Presence.PRESENT) {
+    return "Available - UPnP and NAT-PMP.";
+  }
+  if (portMapping.upnp === Presence.FAILED || portMapping.natpmp === Presence.FAILED) {
+    return "Please try again later.";
+  }
+  return "";
+};
+
+const getCgnatTitle = (presence: Presence) => {
+  if (presence === Presence.FAILED) {
+    return "Failed to determine CGNAT or Double NAT presence";
+  }
+  return "CGNAT or Double NAT";
+};
+
+const getCgnatDescription = (presence: Presence) => {
+  switch (presence) {
+    case Presence.ABSENT:
+      return "Not detected - This is the optimal result.";
+    case Presence.PRESENT:
+      return "Detected (it could also be a VPN) - You may have trouble connecting to other players.";
+    case Presence.FAILED:
+      return `Please try again later.
+        If the failure persists, you can test this in your computer's terminal app.
+        See below:`;
+    default:
+      return "";
+  }
+};
+
+export const NetworkDiagnosticsButton = React.memo(() => {
+  const [dialogOpen, setDialogOpen] = React.useState(false);
+  const [ipAddress, setIpAddress] = React.useState("");
+  const [natType, setNatType] = React.useState(NatType.UNKNOWN);
+  const [portMapping, setPortMapping] = React.useState({
+    upnp: Presence.UNKNOWN,
+    natpmp: Presence.UNKNOWN,
+  } as PortMapping);
+  const [cgnat, setCgnat] = React.useState(Presence.UNKNOWN);
+
+  const runNetworkDiagnostics = async () => {
+    const { address, cgnat, natType, portMapping } = await window.electron.common.runNetworkDiagnostics();
+    setIpAddress(address);
+    setCgnat(cgnat);
+    setNatType(natType);
+    setPortMapping(portMapping);
+  };
+
+  const openDialog = async () => {
+    setIpAddress("");
+    setNatType(NatType.UNKNOWN);
+    setPortMapping({ upnp: Presence.UNKNOWN, natpmp: Presence.UNKNOWN });
+    setCgnat(Presence.UNKNOWN);
+    setDialogOpen(true);
+    await runNetworkDiagnostics();
+  };
```

Comment:
```suggestion
type NetworkInformation = {
  address: string,
  cgnat: Presence,
  natType: NatType,
  portMapping: PortMapping,
}

export const NetworkDiagnosticsButton = React.memo(() => {
  const [dialogOpen, setDialogOpen] = React.useState(false);
  const [isLoading, setIsLoading] = React.useState(false);
  const [isError, setIsError] = React.useState(false);
  const [networkInfo, setNetworkInfo] = React.useState<NetworkInformation | undefined>(undefined);
  
  const runNetworkDiagnostics = async () => {
    setIsLoading(true);
    setIsError(false);
    await window.electron.common.runNetworkDiagnostics().then(setNetworkInfo).catch(() => {
      // Probably want to handle this error state somehow with an error alert.
      setIsError(true);
    }).finally(() => setIsLoading(false));
  };

  const openDialog = async () => {
    setDialogOpen(true);
    await runNetworkDiagnostics();
  };
```

You're conflating the loading state with unknown network type. We should be handling the loading state (and error state technically) explicitly. Since all the data is being fetched at once, again we do not need individual loading states for each section (unless they're doing their own async calls).